frontend/src/components/assessments/assessment-creation/AssessmentEditorHeader.tsx

```diff
@@ -40,23 +43,37 @@ const AssessmentEditorHeader = ({
   name,
   handleSubmit,
   isEditing,
+  onConfirmArchive,
   onConfirmPublish,
+  onDelete,
   onSave,
   onError,
   validateForm,
 }: AssessmentEditorHeaderProps): React.ReactElement => {
   const history = useHistory();
   const [showPublishModal, setShowPublishModal] = React.useState(false);
   const [showDeleteModal, setShowDeleteModal] = React.useState(false);
+  const [showArchiveModal, setShowArchiveModal] = React.useState(false);
+
   const onPublish = () => {
     if (validateForm()) {
       setShowPublishModal(true);
     }
   };
 
+  const onArchive = () => {
+    if (validateForm()) {
+      setShowArchiveModal(true);
+    }
+  };
+
   const handleSave = handleSubmit(onSave, onError);
   const handlePublish = handleSubmit(onPublish, onError);
   const handleConfirmPublish = handleSubmit(onConfirmPublish, onError);
+  const handleArchive = handleSubmit(onArchive, onError);
+  const handleConfirmArchive = handleSubmit(onConfirmArchive, onError);
+  const handleDelete = handleSubmit(onDelete, onError);
+  const handleCancel = () => history.goBack();
```

Comment:
nit: can we call it `handleCloseEditor` or something like that, since there is also a cancel button on the `DeleteModal`?